src/jsi/jsi_class.hpp

```diff
@@ -463,13 +483,20 @@ class ObjectWrap {
 
     static JsiObj create_instance(JsiEnv env, Internal* ptr = nullptr)
     {
-        auto proto = (*s_ctor)->getPropertyAsObject(env, "prototype");
-        auto obj = ObjectCreate(env, proto);
+        if (REALM_UNLIKELY(!s_proto)) {
+            s_proto = (*s_ctor)->getPropertyAsObject(env, "prototype");
+        }
+
+        auto obj = ObjectCreate(env, *s_proto);
+
         set_internal(env, obj, ptr);
 
-        auto wrapper = (*s_ctor)->getProperty(env, "_proxyWrapper");
-        if (!wrapper.isUndefined()) {
-            obj = env(wrapper.asObject(env).asFunction(env).call(env, std::move(obj.get()))).asObject();
+        if (REALM_UNLIKELY(!s_wrapper)) {
+            s_wrapper = (*s_ctor)->getProperty(env, "_proxyWrapper");
+        }
+
+        if (!(*s_wrapper).isUndefined()) {
```

Comment:
Given that we are creating `_proxyWrapper` in C++, we might as well drop it in `s_wrapper` (which could be an `optional<Function>`) when we create it (if it is needed). We could then just do `if (s_wrapper)` which will be slightly faster since it won't need to call into the engine to ask if it is undefined.